aten/src/ATen/native/cuda/UniqueCub.cu

```diff
@@ -0,0 +1,201 @@
+#include <ATen/native/cuda/UniqueCub.cuh>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/cub.cuh>
+
+namespace at {
+namespace native {
+namespace internal {
+
+namespace {
+
+template <typename scalar_t>
+__global__ void adjacent_difference_kernel(
+    int64_t n,
+    const scalar_t* input,
+    int* output) {
+  auto begin = blockIdx.x * blockDim.x + threadIdx.x;
+  auto stride = gridDim.x * blockDim.x;
+  for (auto i = begin; i < n; i += stride) {
+    output[i] = i > 0 ? input[i] != input[i - 1] : 0;
+  }
+}
+
+__global__ void scatter_kernel(
+    int64_t n,
+    const int64_t* input,
+    const int64_t* indices,
+    int64_t* output) {
+  auto begin = blockIdx.x * blockDim.x + threadIdx.x;
+  auto stride = gridDim.x * blockDim.x;
+  for (auto i = begin; i < n; i += stride) {
```

Comment:
Consider using `CUDA_KERNEL_LOOP ` in https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/cuda/detail/KernelUtils.h#L21